api/src/messaging/consumers.py

```diff
@@ -50,128 +51,33 @@ async def receive(self, text_data=None, byte_data=None):
-        await database_sync_to_async(
-            lambda: message_object.tagged_users.add(*tagged_users)
-        )()
-        return {
-            "type": "send_message",
-            "id": message_object.id,
-            "message_body": message_body,
-            "creator": creator,
-            "previous_message": previous_message,
-            "datetime_created": message_object.datetime_created.strftime(
-                "%Y-%m-%dT%H:%M:%S.%fZ"
-            ),
-            "tagged_users": tagged_users,
-        }
-
     # Sends a received message to the group
     async def send_like(self, event):
-        message_id = event["message_id"]
-        is_liked = event["like"]
-        liked_user_id = event["user_id"]
-        user_id = self.scope["user"].id
-
-        if self.member["id"] == user_id:
-            await self.send(
-                text_data=json.dumps(
-                    {
-                        "message_id": message_id,
-                        "like": is_liked,
-                        "user_id": liked_user_id,
-                    }
-                )
-            )
-
-        else:
-            await self.channel_layer.group_discard(
-                self.room_group_name, self.channel_name
-            )
+        data = {
+            "message_id": event["message_id"],
+            "like": event["like"],
+            "user_id": event["user_id"],
+        }
+        await create_data(self, data)
 
     async def send_message(self, event):
-        message_id = event["id"]
-        message_body = event["message_body"]
-        creator = event["creator"]
-        datetime_created = event["datetime_created"]
-        previous_message = event["previous_message"]
-        tagged_users = event["tagged_users"]
-        user_id = self.scope["user"].id
-
-        if self.member["id"] == user_id:
-            await self.send(
-                text_data=json.dumps(
-                    {
-                        "id": message_id,
-                        "message_body": message_body,
-                        "creator": creator,
-                        "previous_message": previous_message,
-                        "datetime_created": datetime_created,
-                        "tagged_users": tagged_users,
-                    }
-                )
-            )
-
-        else:
-            await self.channel_layer.group_discard(
-                self.room_group_name, self.channel_name
-            )
+        data = {
+            "id": event["id"],
+            "message_body": event["message_body"],
+            "creator": event["creator"],
+            "previous_message": event["previous_message"],
+            "datetime_created": event["datetime_created"],
+            "tagged_users": event["tagged_users"],
+        }
+        await create_data(self, data)
+
+
+# Helper function for creating data
+async def create_data(consumer, data):
+    user_id = consumer.scope["user"].id
+    if consumer.member["id"] == user_id:
```

Comment:
Since we aren't reusing the user_id variable anywhere else, we might as well just use consumer.scope["user"].id directly in the if condition.